sdk/samples/iot/sample_pnp_component_mqtt.c

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "sample_pnp_component_mqtt.h"
+
+#include <stdint.h>
+
+#include <azure/core/az_span.h>
+
+static int32_t request_id_int;
+static char request_id_buf[8];
+
+// Create request id span which increments request id integer each call. Capable of holding 8 digit
+// number.
+az_span get_request_id(void)
+{
+  az_span remainder;
+  az_span out_span = az_span_init((uint8_t*)request_id_buf, sizeof(request_id_buf));
+  az_result result = az_span_i32toa(out_span, request_id_int++, &remainder);
+  (void)remainder;
+  (void)result;
```

Comment:
What happens if `request_id_int` becomes too large and won't fit in a buffer of size 8? We are ignoring `AZ_ERROR_INSUFFICIENT_SPAN_SIZE`.

`2147483647` and `-2147483648` are length 10 and 11 respectively.

Maybe add a comment on why it is OK for us to ignore remainder and result.

Writing 100, and then 1, without clearing/accounting for reminder would have unexpected results. Our usage here works only because we are always incrementing request_id by one.

Reply:
Yea I can add some comments on the assumptions we are making. In this case, if this overflows, this sample has been running for a LONGGGGGGG time.

And yes about the last point but incrementing by one each time gets around that.

Reply:
actually I'm going to change it to a `uint32_t` so if it does overflow it doesn't matter. If it does overflow then the message with `rid = 0` is long gone.

Also, how does 
```c
return az_span_slice(out_span, 0, (int32_t)(az_span_ptr(remainder) - az_span_ptr(out_span)));
```
sound to make sure that just the digits are returned?

Reply:
That sounds good. But use `az_span_size`.